core/src/main/scala/org/http4s/Media.scala

```diff
@@ -1,65 +1,36 @@
 package org.http4s
 
-import cats.MonadError
-import cats.implicits._
+// import cats.MonadError
+// import cats.implicits._
 import fs2.Stream
 import fs2.text.utf8Decode
 import org.http4s.headers._
 import org.http4s.util.decode
 
-trait Media[F[_]] {
-  def body: EntityBody[F]
-  def headers: Headers
-  def covary[F2[x] >: F[x]]: Media[F2]
+trait Media[M[_[_]]]{
```

Comment:
Looks like `Media` is agnostic of exact `M` structure (i.e. it doesn't use the fact that it is parameterized by `F`) - so there is no need in *too-higher-kinded* type `M[_[_]]`.
May I suggest you to consider `trait Media[F[_], -T]`, so you can have implicit instances like `Media[F, Part[F]]` and `Media[F, M <: Message[F]]`?

Reply:
Moreover, it is possible to hide `F[_]` to a type member, so you would get something like haskell's associated types:

```scala
trait Media[T] {
  type F[_]

  def body(of: T): EntityBody[F]
  def headers(of: T): Headers
}

implicit def mediaPartInstance[S[_]]: Media[Part[S]] = new Media[Part[S]] {
  type F[a] = S[a]

  def body(of: T): EntityBody[F] = ???
  def headers(of: T): Headers = ??
}
```

Reply:
Contravariant typeclasses are a bit unusual.  Is it just to support `Message`?  If that's all, I could see either eliminating `Message` or making it an extension of the `Media` typeclass.

Reply:
@rossabaker ah, you are right, no need in contravariance here. It still should be possible to write instance for all subclasses of `Message[F]` without it.